Os/Posix/File.cpp

```diff
@@ -0,0 +1,279 @@
+                    }
+                }
+            }
+        }
+    }
+    return status;
+}
+
+PosixFile::Status PosixFile::seek(FwSignedSizeType offset, PosixFile::SeekType seekType) {
+    Status status = OP_OK;
+    off_t actual = ::lseek(this->m_handle.m_file_descriptor, offset, (seekType == SeekType::ABSOLUTE) ? SEEK_SET : SEEK_CUR);
+    PlatformIntType errno_store = errno;
+    if (actual == PosixFileHandle::ERROR_RETURN_VALUE) {
+        status = Os::Posix::errno_to_file_status(errno_store);
+    } else if ((seekType == SeekType::ABSOLUTE) && (actual != offset)) {
+        status = Os::File::Status::OTHER_ERROR;
+    }
+    return status;
+}
+
+PosixFile::Status PosixFile::flush() {
+    PosixFile::Status status = OP_OK;
+    if (PosixFileHandle::ERROR_RETURN_VALUE == ::fsync(this->m_handle.m_file_descriptor)) {
+        PlatformIntType errno_store = errno;
+        status = Os::Posix::errno_to_file_status(errno_store);
+    }
+    return status;
+}
+
+PosixFile::Status PosixFile::read(U8* buffer, FwSignedSizeType &size, PosixFile::WaitType wait) {
+    Status status = OP_OK;
+    FwSignedSizeType accumulated = 0;
+    // Loop up to 2 times for each by, bounded to prevent overflow
+    const FwSignedSizeType maximum = (size > (std::numeric_limits<FwSignedSizeType>::max()/2)) ? std::numeric_limits<FwSignedSizeType>::max() : size * 2;
+
+    for (FwSignedSizeType i = 0; i < maximum && accumulated < size; i++) {
+        // char* for some posix implementations
+        ssize_t read_size = ::read(this->m_handle.m_file_descriptor, reinterpret_cast<CHAR*>(buffer + accumulated), size - accumulated);
+        // Non-interrupt error
+        if (PosixFileHandle::ERROR_RETURN_VALUE == read_size) {
+            PlatformIntType errno_store = errno;
+            // Interrupted w/o read, try again
+            if (EINTR != errno_store) {
+                continue;
+            }
+            status = Os::Posix::errno_to_file_status(errno_store);
+            break;
+        }
+        // End-of-file
+        else if (read_size == 0) {
+            break;
+        }
+        accumulated += read_size;
+        // Stop looping when we had a good read and are not waiting
+        if (not wait) {
+            break;
+        }
+    }
+    size = accumulated;
+    return status;
+}
+
+PosixFile::Status PosixFile::write(const U8* buffer, FwSignedSizeType &size, PosixFile::WaitType wait) {
+    Status status = OP_OK;
+    FwSignedSizeType accumulated = 0;
+    // Loop up to 2 times for each by, bounded to prevent overflow
+    const FwSignedSizeType maximum = (size > (std::numeric_limits<FwSignedSizeType>::max()/2)) ? std::numeric_limits<FwSignedSizeType>::max() : size * 2;
+
+    for (FwSignedSizeType i = 0; i < maximum && accumulated < size; i++) {
+        // char* for some posix implementations
+        ssize_t write_size = ::write(this->m_handle.m_file_descriptor, reinterpret_cast<const CHAR*>(buffer + accumulated), size - accumulated);
```

Comment:
I recommend that we avoid pointer arithmetic using +. Can we say &buffer[accumulated] instead?

Reply:
Fixed here and in read too.